File: test/system/hello_world_mpi.cpp
#include <iostream>
#include <mpi.h>

using namespace std;

int main (int argc, char *argv[])
{
  int rank, size;
  MPI_Init (&argc, &argv);
  MPI_Comm_rank (MPI_COMM_WORLD, &rank);
  MPI_Comm_size (MPI_COMM_WORLD, &size);
  if (rank==0)
    cout << size << "." << rank << ": hello world" << endl;
  MPI_Finalize();
  return 0;
} 

File: test/system/hello_world_mpi.f
      program hello_world_mpi
      include 'mpif.h'
      integer rank, size, ierror, tag, status(MPI_STATUS_SIZE)
      call MPI_INIT(ierror)
      call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierror)
      call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierror)
      if (rank .eq. 0) then
         write(*,'(I6,A,I1,A)') size, '.', rank, ': hello world'
      end if
      call MPI_FINALIZE(ierror)
      
      end program hello_world_mpi
      
